packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/lambda-walkthrough.ts

```diff
@@ -172,14 +172,15 @@ export async function updateWalkthrough(context, lambdaToUpdate?: string) {
   const cfnContent = context.amplify.readJsonFile(cfnFilePath);
 
   // check for layer parameters if not added
-  functionParameters.dependsOn.forEach(val => {
-    const param: string = `${val.category}${val.resourceName}Arn`;
-    // apend the cloudwatch parameters if not present
-    if (cfnContent.Parameters[`${param}`] === undefined) {
-      cfnContent.Parameters[`${param}`] = {
-        Type: 'String',
-        Default: `${param}`,
-      };
+  functionParameters.lambdaLayers.forEach(val => {
+    if ((val as ProjectLayer).resourceName) {
+      const param: string = `function${(val as ProjectLayer).resourceName}Arn`;
+      if (cfnContent.Parameters[`${param}`] === undefined) {
+        cfnContent.Parameters[`${param}`] = {
+          Type: 'String',
+          Default: `${param}`,
+        };
+      }
```

Comment:
I would write something like this. I think using a resourceName const makes this a bit more readable.

```ts
const resourceName = _.get((val as ProjectLayer), ['resourceName'], null);
if (resourceName) {
  const param: string = `function${resourceName}Arn`;
  if (cfnContent.Parameters[`${param}`] === undefined) {
    cfnContent.Parameters[`${param}`] = { Type: 'String', Default: `${param}` };
  }
}
```

Reply:
Nice suggestion!!